dtfs-central-api/api-tests/v1/bank/banks-get.api-test.ts

```diff
@@ -0,0 +1,107 @@
+    it('responds with a 200 (Ok) and all banks without reporting years', async () => {
+      // Arrange
+      const requestUrl = axios.getUri({
+        url: BASE_URL,
+        params: { includeReportingYears },
+      });
+
+      const banks = MOCK_BANKS.map((bank) => ({ ...bank, _id: bank._id.toString() }));
+
+      // Act
+      const response = await testApi.get(requestUrl);
+
+      // Assert
+      expect(response.status).toBe(axios.HttpStatusCode.Ok);
+      expect(response.body).toEqual(banks);
+    });
+  });
+
+  describe("when the 'includeReportingYears' query is set to 'true'", () => {
+    const reportIdGenerator = idGenerator();
+
+    const aUtilisationReportForStatusBankIdAndMonthlyReportPeriod = (
+      status: UtilisationReportReconciliationStatus,
+      bankId: string,
+      monthAndYear: MonthAndYear,
+    ): UtilisationReportEntity =>
+      UtilisationReportEntityMockBuilder.forStatus(status)
+        .withId(reportIdGenerator.next().value)
+        .withBankId(bankId)
+        .withReportPeriod({
+          start: monthAndYear,
+          end: monthAndYear,
+        })
+        .build();
+
+    it('responds with a 200 (Ok) and all banks with reporting years', async () => {
+      // Arrange
+      const requestUrl = axios.getUri({
+        url: BASE_URL,
+        params: { includeReportingYears: true },
+      });
+
+      const utilisationReports = [
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_ONE, { month: 1, year: 2021 }),
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_ONE, { month: 2, year: 2021 }),
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_ONE, { month: 2, year: 2022 }),
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_TWO, { month: 1, year: 2020 }),
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('REPORT_NOT_RECEIVED', BANK_ID_TWO, { month: 1, year: 2022 }), // report exists but was not submitted
+        aUtilisationReportForStatusBankIdAndMonthlyReportPeriod('PENDING_RECONCILIATION', BANK_ID_TWO, { month: 1, year: 2023 }),
+      ];
+      await SqlDbHelper.saveNewEntries('UtilisationReport', utilisationReports);
+
+      const banksWithReportingYears = [
+        { ...MOCK_BANKS[0], _id: MOCK_BANKS[0]._id.toString(), reportingYears: [2021, 2022] },
+        { ...MOCK_BANKS[1], _id: MOCK_BANKS[1]._id.toString(), reportingYears: [2020, 2023] },
+      ];
+
+      // Act
+      const response = await testApi.get(requestUrl);
+
+      // Assert
+      expect(response.status).toBe(axios.HttpStatusCode.Ok);
+      expect(response.body).toEqual(banksWithReportingYears);
+    });
+  });
+
+  function* idGenerator(): Generator<number, number, unknown> {
+    let id = 1;
+    while (true) {
+      yield id;
+      id += 1;
+    }
+  }
```

Comment:
I can see this is a duplicate in at least 4x places now, could we move to a DRY test helper?

Reply:
The specifics of an `idGenerator` could vary depending on what each file wants to test. For instance, here we are specifically testing ids which are positive integers, but someone may mistakenly import this to their test file and expect that it generates some other form of an "id" (`ObjectId`, for instance) which they would then need to check the actual implementation of `idGenerator` in order to see what was going wrong with their test

Reply:
I have added a test helper called `getIncrementingPositiveIntegerIdGenerator`